kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/framework/concurrency/continually.kt

```diff
@@ -0,0 +1,57 @@
+package io.kotest.framework.concurrency
+
+import io.kotest.assertions.failure
+import io.kotest.common.ExperimentalKotest
+import io.kotest.mpp.timeInMillis
+import kotlinx.coroutines.delay
+
+@ExperimentalKotest
+data class ContinuallyState(val start: Millis, val end: Millis, val times: Int)
+
+@ExperimentalKotest
+fun interface ContinuallyListener<in T> {
+   fun onEval(t: T, state: ContinuallyState)
+
+   companion object {
+      val default = ContinuallyListener<Any?> { _, _ -> }
+   }
+}
+
+@ExperimentalKotest
+suspend fun <T> continually(
+   duration: Millis,
+   interval: Interval = 25L.fixed(),
+   listener: ContinuallyListener<T> = ContinuallyListener.default,
+   f: ConcurrencyProducer<T>
+): T? = continually(PatienceConfig(duration, interval), listener, f)
+
+@ExperimentalKotest
+suspend fun <T> continually(
```

Comment:
should we do the same thing with a continually builder

Reply:
done